modules/somoaudienceBidAdapter.js

```diff
@@ -0,0 +1,133 @@
+import {logError, getTopWindowLocation} from 'src/utils';
+import { registerBidder } from 'src/adapters/bidderFactory';
+
+export const spec = {
+
+  code: 'somoaudience',
+
+  aliases: ['somo'],
+
+  supportedMediaTypes: ['banner'],
+
+  isBidRequestValid: bid => (
+    !!(bid && bid.params && bid.params.placementId)
+  ),
+
+  buildRequests: function(bidRequests) {
+    logError('bidRequests', 'ERROR', bidRequests);
+    return bidRequests.map(bidRequest => {
+      return {
+        method: 'POST',
+        url: '//publisher-east.mobileadtrading.com/rtb/bid?s=' + bidRequest.params.placementId.toString(),
+        data: openRtbRequest(bidRequest),
+        bidRequest: bidRequest
+      };
+    });
+  },
+
+  interpretResponse(response, request) {
+    return bidResponseAvailable(request, response);
+  }
+};
+
+function bidResponseAvailable(bidRequest, bidResponse) {
+  let bidResponses = [];
+  let bidId = 1;
+  if(typeof bidRequest != 'undefined' && typeof bidRequest.bidRequest != 'undefined' && typeof bidRequest.bidRequest.bidId != 'undefined') {
+    bidId = bidRequest.bidRequest.bidId;
+  }
+  if (bidResponse) {
```

Comment:
your check should be `bidResponse.body`.  You will always receive this object https://github.com/prebid/Prebid.js/blob/master/src/adapters/bidderFactory.js#L284